resources/fieldsets/call_to_action.yaml

```diff
@@ -0,0 +1,112 @@
+title: 'Call to action'
+fields:
+  -
+    import: component_settings
+  -
+    handle: title_heading
+    field:
+      input_type: text
+      antlers: false
+      type: text
+      display: 'Title - Heading'
+      icon: text
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+      width: 50
+  -
+    handle: title_subheading
+    field:
+      input_type: text
+      antlers: false
+      type: text
+      display: 'Title - Subheading'
+      icon: text
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+      width: 50
+  -
+    handle: logo
+    field:
+      max_files: 1
+      mode: list
+      container: assets
+      restrict: false
+      allow_uploads: true
+      show_filename: true
+      show_set_alt: true
+      type: assets
+      display: Logo
+      icon: assets
+      listable: hidden
+      instructions_position: above
+      visibility: visible
+      replicator_preview: true
+      hide_display: false
+  -
+    handle: buttons
+    field:
+      collapse: false
+      previews: true
+      max_sets: 2
+      fullscreen: true
+      sets:
+        extra_button:
+          display: 'Extra button'
+          instructions: null
+          icon: null
+          sets:
+            button:
+              display: Button
+              instructions: null
+              icon: null
+              fields:
```

Comment:
There's a couple of places where you've manually made the same button, maybe put it in a fieldset?